modules/distr-tracing-otel-migrating-from-jaeger-with-sidecars.adoc

```diff
@@ -0,0 +1,145 @@
+// Module included in the following assemblies:
+//
+// * distr-tracing-otel-migrating.adoc
+
+:_content-type: PROCEDURE
+[id="distr-tracing-otel-migrating-from-jaeger_{context}"]
+= (Temporary title) Migrating from {JaegerName} to {OTELName} with sidecars
+
+.Prerequisites
+
+* Existing instrumentation of applications uses {JaegerName} for tracing.
+* {OTELName} is installed.
+
+.Procedure
+
+The Red Hat Red Hat OpenShift Distributed Tracing Data Collection Operator can inject sidecars into Deployment workloads.
+
+Configure OpenTelemetry Collector as sidecar
+
+----
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: otel
+  namespace: otel-collector-example
+spec:
+  mode: sidecar
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+          thrift_binary:
+          thrift_compact:
+          thrift_http:
+    processors:
+      batch:
+      memory_limiter:
+        check_interval: 1s
+        limit_percentage: 50
+        spike_limit_percentage: 30
+      resourcedetection:
+        detectors: [openshift]
+        timeout: 2s
+    exporters:
+      otlp:
+        endpoint: "tempo-example-gateway:8090"
+        tls:
+          insecure: true
+    service:
+      pipelines:
+        traces:
+          receivers: [jaeger]
+          processors: [memory_limiter, resourcedetection, batch]
+          exporters: [otlp]
+----
+
+Create a Service Account for running your application:
+
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: otel-collector-sidecar
+----
+
+Create a ClusterRole for some permissions needed by some processors
+
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: otel-collector-sidecar
+rules:
+# Permissions for infrastructures and infrastructures/status are needed for resourcedetectionprocessor
+- apiGroups: ["config.openshift.io"]
+  resources: ["infrastructures", "infrastructures/status"]
+  verbs: ["get", "watch", "list"]
+----
+
+Create a ClusterRoleBinding to give those permissions to the Service Account
+
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: otel-collector-sidecar
+subjects:
+- kind: ServiceAccount
+  name: otel-collector-sidecar
+roleRef:
+  kind: ClusterRole
+  name: otel-collector-sidecar
+  apiGroup: rbac.authorization.k8s.io
+----
+
+Deploy the OpenTelemetry Collector as sidecar
+
+----
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: otel
+  namespace: otel-collector-example
+spec:
+  mode: sidecar
```

Comment:
Need to specify the created serviceAccount in the previous step.

```
spec:
  mode: sidecar
  serviceAccount: otel-collector-sidecar
  config: |
```

Reply:
The serviceAccount needs to be set to the application where the sidecar is injected